main.go

```diff
@@ -196,17 +207,25 @@ func (c AzureClient) deleteAADUsers(ctx context.Context, prefix string, actually
 }
 
 func (c AzureClient) deleteResourceGroups(ctx context.Context, numberOfResourceGroupsToDelete int, prefix string, actuallyDelete bool) error {
-	max := int32(numberOfResourceGroupsToDelete)
 	log.Printf("[DEBUG] Loading the first %d resource groups to delete", numberOfResourceGroupsToDelete)
-	groups, err := c.resourcesClient.List(ctx, "", &max)
+
+	subscriptionId := commonids.NewSubscriptionID(c.authClient.SubscriptionID)
+	var listOpts resourcegroups.ListOperationOptions
+	groups, err := c.resourcesClient.List(ctx, subscriptionId, listOpts)
```

Comment:
we have the default one available:

```suggestion
	groups, err := c.resourcesClient.List(ctx, subscriptionId, resourcegroups.DefaultListOperationOptions())
```